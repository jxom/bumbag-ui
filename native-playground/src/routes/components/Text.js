import React from 'react';
import { Box, Text } from 'bumbag-native';
import { Preview } from '../../components/Preview';
import { PreviewSection } from '../../components/PreviewSection';

export default function App() {
  return (
    <Box>
      <Text fontSize="20px" fontWeight="500" marginBottom="16px">
        Text
      </Text>
      <PreviewSection title="Basic">
        <Preview>
          <Box>
            <Text>
              Hello world, this is a bit of <Text fontWeight="bold">text</Text>.
            </Text>
          </Box>
        </Preview>
      </PreviewSection>
      <PreviewSection title="Font weights">
        <Preview>
          <Box>
            <Text fontWeight="normal">Hello world, this is a bit of text.</Text>
            <Text fontWeight="semibold">Hello world, this is a bit of text.</Text>
            <Text fontWeight="bold">Hello world, this is a bit of text.</Text>
            <Text fontWeight="100">Hello world, this is a bit of text.</Text>
            <Text fontWeight="200">Hello world, this is a bit of text.</Text>
            <Text fontWeight="300">Hello world, this is a bit of text.</Text>
            <Text fontWeight="400">Hello world, this is a bit of text.</Text>
            <Text fontWeight="500">Hello world, this is a bit of text.</Text>
            <Text fontWeight="600">Hello world, this is a bit of text.</Text>
            <Text fontWeight="700">Hello world, this is a bit of text.</Text>
            <Text fontWeight="800">Hello world, this is a bit of text.</Text>
            <Text fontWeight="900">Hello world, this is a bit of text.</Text>
          </Box>
        </Preview>
      </PreviewSection>
    </Box>
  );
}
