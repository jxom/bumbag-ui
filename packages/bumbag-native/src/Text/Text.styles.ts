import { Animated, Platform } from 'react-native';
import { styled } from '../styled';
import { getFontStyles } from '../utils/getFontStyles';
import { palette, theme } from '../utils/theme';

function getTextStyles(props) {
  return `
    color: ${palette('text')(props)};

    ${
      Platform.OS === 'web'
        ? `
            font-size: unset;
          `
        : ''
    }

    ${!props.fontWeight ? getFontStyles({ fontWeight: '400' })(props) : ''}

    ${theme('native.Text', 'styles.base')(props)};
  `;
}

export const StyledText = styled.Text`
  ${getTextStyles}
`;

export const StyledAnimatedText = styled(Animated.Text)`
  ${getTextStyles}
`;
