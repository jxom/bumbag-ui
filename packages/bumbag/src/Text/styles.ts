import { cssClass } from '../styled';
import { theme } from '../utils';

export const Text = (styleProps) => cssClass`
  line-height: 1.2;

  abbr& {
    border-bottom: 1px dotted;
    cursor: help;
    text-decoration: none;
  }

  code& {
    font-family: monospace;
  }

  i& {
    font-style: italic;
  }

  kbd& {
    background: #454d5d;
    border-radius: 0.1rem;
    color: #fff;
    fill: #fff;
    padding: 0.1rem 0.2rem;
  }

  mark& {
    background: #ffe9b3;
    border-radius: 0.1rem;
    padding: 0.1rem 0.2rem;
  }

  strong& {
    font-weight: bold;
  }

  sub&,
  sup& {
    font-size: 0.8em;
    vertical-align: baseline;
  }

  ins&,
  u& {
    text-decoration-skip: ink edges;
  }

  i&,
  cite&,
  var&,
  em& {
    font-style: italic;
  }

  kbd&,
  pre&,
  samp& {
    font-family: 'SF Mono', 'Segoe UI Mono', 'Roboto Mono', Menlo, Courier, monospace;
  }

  & .bb-Icon {
    vertical-align: -0.125em;
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
