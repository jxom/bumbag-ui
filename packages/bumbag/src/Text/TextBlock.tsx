import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalTextBlockProps = {};
export type TextBlockProps = BoxProps & LocalTextBlockProps;

const useProps = createHook<TextBlockProps>(
  (props, { themeKey, themeKeyOverride }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.TextBlock,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'Text.Block' }
);

export const TextBlock = createComponent<TextBlockProps>(
  (props) => {
    const textBlockProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: textBlockProps });
  },
  {
    attach: {
      useProps,
      displayName: 'Text.Block',
    },
    defaultProps: {
      use: 'span',
    },
    themeKey: 'Text.Block',
  }
);
