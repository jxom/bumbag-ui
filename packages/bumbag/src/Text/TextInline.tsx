import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalTextInlineProps = {};
export type TextInlineProps = BoxProps & LocalTextInlineProps;

const useProps = createHook<TextInlineProps>(
  (props, { themeKey, themeKeyOverride }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.TextInline,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'Text.Inline' }
);

export const TextInline = createComponent<TextInlineProps>(
  (props) => {
    const textInlineProps = useProps(props);
    return createElement({
      children: props.children,
      component: ReakitBox,
      use: props.use,
      htmlProps: textInlineProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'Text.Inline',
    },
    defaultProps: {
      use: 'span',
    },
    themeKey: 'Text.Inline',
  }
);
