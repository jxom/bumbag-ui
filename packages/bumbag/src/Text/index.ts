import * as styles from './styles';
import { Text as _Text } from './Text';
import { TextBlock } from './TextBlock';
import { TextInline } from './TextInline';

export * from './Text';
export const Text = Object.assign(_Text, {
  Block: TextBlock,
  Inline: TextInline,
});
export { styles as textStyles };
