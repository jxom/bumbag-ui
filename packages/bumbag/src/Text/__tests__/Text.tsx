import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Text } from '../Text';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Text>Hello world</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Text color="primary">Hello world</Text>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Text use="p">Hello world</Text>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Text props', () => {
      const { result } = renderHook(() => Text.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Text>{(TextProps) => <div {...TextProps}>Hello world</div>}</Text>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Text.styles.base should render correctly', () => {
    const { container } = render(<Text>hello world</Text>, {
      theme: { Text: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Text>hello world</Text>, {
      theme: { Text: { defaultProps: { className: 'test', color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
