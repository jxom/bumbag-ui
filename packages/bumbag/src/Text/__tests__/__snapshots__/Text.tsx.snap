// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

abbr.emotion-0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

code.emotion-0 {
  font-family: monospace;
}

i.emotion-0 {
  font-style: italic;
}

kbd.emotion-0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.emotion-0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.emotion-0 {
  font-weight: bold;
}

sub.emotion-0,
sup.emotion-0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.emotion-0,
u.emotion-0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.emotion-0,
cite.emotion-0,
var.emotion-0,
em.emotion-0 {
  font-style: italic;
}

kbd.emotion-0,
pre.emotion-0,
samp.emotion-0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.emotion-0 .bb-Icon {
  vertical-align: -0.125em;
}

<p
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Text"
>
  Hello world
</p>
`;

exports[`composition hook should return with Text props 1`] = `
Object {
  "className": "css-1jrn33c css-63it0p css-jwhr8m bb-Box bb-Text",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

abbr.emotion-0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

code.emotion-0 {
  font-family: monospace;
}

i.emotion-0 {
  font-style: italic;
}

kbd.emotion-0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.emotion-0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.emotion-0 {
  font-weight: bold;
}

sub.emotion-0,
sup.emotion-0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.emotion-0,
u.emotion-0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.emotion-0,
cite.emotion-0,
var.emotion-0,
em.emotion-0 {
  font-style: italic;
}

kbd.emotion-0,
pre.emotion-0,
samp.emotion-0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.emotion-0 .bb-Icon {
  vertical-align: -0.125em;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Text"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

abbr.emotion-0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

code.emotion-0 {
  font-family: monospace;
}

i.emotion-0 {
  font-style: italic;
}

kbd.emotion-0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.emotion-0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.emotion-0 {
  font-weight: bold;
}

sub.emotion-0,
sup.emotion-0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.emotion-0,
u.emotion-0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.emotion-0,
cite.emotion-0,
var.emotion-0,
em.emotion-0 {
  font-style: italic;
}

kbd.emotion-0,
pre.emotion-0,
samp.emotion-0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.emotion-0 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-2 {
  color: var(--bb-palette-primary,#000000) !important;
}

<span
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-Text"
>
  hello world
</span>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

abbr.emotion-0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

code.emotion-0 {
  font-family: monospace;
}

i.emotion-0 {
  font-style: italic;
}

kbd.emotion-0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.emotion-0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.emotion-0 {
  font-weight: bold;
}

sub.emotion-0,
sup.emotion-0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.emotion-0,
u.emotion-0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.emotion-0,
cite.emotion-0,
var.emotion-0,
em.emotion-0 {
  font-style: italic;
}

kbd.emotion-0,
pre.emotion-0,
samp.emotion-0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.emotion-0 .bb-Icon {
  vertical-align: -0.125em;
}

<span
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Text"
>
  Hello world
</span>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

abbr.emotion-0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

code.emotion-0 {
  font-family: monospace;
}

i.emotion-0 {
  font-style: italic;
}

kbd.emotion-0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.emotion-0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.emotion-0 {
  font-weight: bold;
}

sub.emotion-0,
sup.emotion-0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.emotion-0,
u.emotion-0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.emotion-0,
cite.emotion-0,
var.emotion-0,
em.emotion-0 {
  font-style: italic;
}

kbd.emotion-0,
pre.emotion-0,
samp.emotion-0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.emotion-0 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-2 {
  color: var(--bb-palette-primary,#000000) !important;
}

<span
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Text"
>
  Hello world
</span>
`;

exports[`theming Text.styles.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

abbr.emotion-0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

code.emotion-0 {
  font-family: monospace;
}

i.emotion-0 {
  font-style: italic;
}

kbd.emotion-0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.emotion-0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.emotion-0 {
  font-weight: bold;
}

sub.emotion-0,
sup.emotion-0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.emotion-0,
u.emotion-0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.emotion-0,
cite.emotion-0,
var.emotion-0,
em.emotion-0 {
  font-style: italic;
}

kbd.emotion-0,
pre.emotion-0,
samp.emotion-0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.emotion-0 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-0 {
  background-color: var(--bb-palette-red,#ff0000);
}

<span
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Text"
>
  hello world
</span>
`;
